crates/cargo-util/src/paths.rs

```diff
@@ -823,6 +857,30 @@ mod tests {
         assert_eq!(contents, original_contents);
     }
 
+    #[test]
+    #[cfg(unix)]
+    fn write_atomic_permissions() {
```

Comment:
Could we have this test in the first commit showing the problematic behavior, and the next commit fixes both the test and the behavior? By doing so it's a bit clearer to reviewers to just read the diff and understand what has been changed.

Reply:
Reorganized it so that the first commit adds the tests, and the second commit adds the `write_atomic` changes.

Reply:
As mentioned in one of the other threads, I fixed this so that the first commit adds the test for the existing behaviour, and the second commit fixes the behaviour and updates the test for the new behaviour.